arch/arm/src/stm32/stm32f20xxf40xx_flash.c

```diff
@@ -226,83 +279,65 @@ int stm32_flash_writeprotect(size_t page, bool enabled)
       up_waste();
     }
 
-  /* Relock options */
+  /* Re-lock options */
 
   modifyreg32(STM32_FLASH_OPTCR, 0, FLASH_OPTCR_OPTLOCK);
   return 0;
 }
 
 size_t up_progmem_pagesize(size_t page)
 {
-  static const size_t page_sizes[STM32_FLASH_NPAGES] = STM32_FLASH_SIZES;
-
-  if (page >= sizeof(page_sizes) / sizeof(*page_sizes))
-    {
-      return 0;
-    }
-  else
-    {
-      return page_sizes[page];
-    }
+  return PAGESIZE;
```

Comment:
This will not work on the F4 the page (sector) size is not constant.

Reply:
Hi David, 
this is a naming issue. Commit 0f18e8cc328f1a420135531eb2afe57b27d3332f introduced 'page' as the unit for r/w access. This is constant for the F4 (128 bit). The not constant sector size is only relevant for erase operations and therefore accessible via up_progmem_erasesize().

Reply:
@bjbrandt Maybe I should ask how this is used?  What happens if I call up_progmem_ispageerased(0), it use to tell me that the sector was erased.  Then it would be used as

```
	/* blank-check the sector */

	bool blank = up_progmem_ispageerased(sector) == 0;

	/* erase the sector if it failed the blank check */
	if (!blank) {
		up_progmem_eraseblock(sector);
```
How would this be done wit these changes?

Reply:
@davids5 For this purpose it would be helpful to have a function like up_progmem_iseraseblockerased() or up_progmem_issectorerased(), I guess. As long as there is no such function defined in progmem.h you need to iterate over all pages of the desired sector instead.